byzcoin/statetrie.go

```diff
@@ -149,13 +155,84 @@ func (t *stagingStateTrie) GetVersion() Version {
 	return readVersion(t)
 }
 
+func (t *stagingStateTrie) LoadConfigFromTrie() (*ChainConfig, error) {
+	return t.loadConfigFromTrie(t)
+}
+
+func (t *stagingStateTrie) LoadDarcFromTrie(id darc.ID) (*darc.Darc, error) {
+	return t.loadDarcFromTrie(t, id)
+}
+
+type trieCache struct {
+	config *ChainConfig
+	darcs  map[string]*darc.Darc
+}
+
+func (tc *trieCache) loadConfigFromTrie(st ReadOnlyStateTrie) (*ChainConfig, error) {
+	if tc.config != nil {
+		return tc.config, nil
```

Comment:
Depending on how usual it is to retrieve the config, one can move it at initialisation time of the `trieCache`.

Reply:
premature optimisation ;) sometimes the tries are initialized just with the structure, so I'd have to replace all those with a `NewTrie`, too...